src/os_hacks.rs

```diff
@@ -45,3 +46,78 @@ pub fn setup_python_deps_on_ubuntu_18_04(pyroot: &Path, pkg_versions: &[&str]) -
 
     Ok(())
 }
+
+/// This is the final step of setup, where we look for OSes that require additional setup steps
+/// beyond the usual ones that we have done already.
+pub fn setup_os_hacks(kani_dir: &Path, os: &Info) -> Result<()> {
+    match os.os_type() {
+        os_info::Type::NixOS => setup_nixos_patchelf(kani_dir),
+        os_info::Type::Linux => {
+            // NixOs containers are detected as Unknown Linux, so use a fallback hack:
+            if std::env::var_os("NIX_CC").is_some() && Path::new("/etc/nix").exists() {
+                return setup_nixos_patchelf(kani_dir);
+            }
+            Ok(())
+        }
+        _ => Ok(()),
+    }
+}
+
+/// On NixOS, the dynamic linker does not live at the standard path, and so our downloaded
+/// pre-built binaries need patching.
+/// In addition, the C++ standard library (needed by the CBMC binaries we ship) also does not
+/// have a standard path, and so we need to inject an rpath into those binaries to get them
+/// to successfully link at runtime.
```

Comment:
I'm assuming LD_LIBRARY_PATH didn't do the trick, right?

Reply:
yeah `patchelf` is nice since it makes it "just work" every time you invoke the command, rather than having to remember to correctly set up the LD_LIBRARY_PATH

Reply:
If the path never changes, I agree that this is a more permanent solution, although a bit invasive.

Reply:
cargo-bolero was actually recently added (not even by me which is cool) so it's pretty easy to get packages in there: https://github.com/NixOs/nixpkgs/blob/master/pkgs/development/tools/rust/cargo-bolero/default.nix

Reply:
That's cool!

In the meantime, if the path changes, we should probably use LD_LIBRARY_PATH or re-patch the binaries if the value has changed. We don't expect users to invoke `kani-driver` or `kani-compiler` directly, so we might as well ensure that they always work.

Reply:
I could be wrong, but the annoying part is that our users will have to re-install `kani-verifier` crate, but that won't fix the problem since the bundle we downloaded will still be on the disk. So when they run `kani`, they will get another crash now because kani-driver is broken, right? At that point, they will have to run `kani setup` to force everything to be reinstalled.

Not a blocker since nothing works today, but we should at least document that.

Reply:
Yep that's correct. But what's here is definitely an improvement. Like I said, I think for the best user experience is to get into nixpkgs.